src/main/java/com/questhelper/quests/treegnomevillage/TreeGnomeVillage.java

```diff
@@ -0,0 +1,207 @@
+		notCompleteSecondTracker = new VarbitCondition(TRACKER_2_VARBITID, 0);
+		notCompleteThirdTracker = new VarbitCondition(TRACKER_3_VARBITID, 0);
+
+		completeFirstTracker = new VarbitCondition(TRACKER_1_VARBITID, 1);
+		completeSecondTracker = new VarbitCondition(TRACKER_2_VARBITID, 1);
+		completeThirdTracker = new VarbitCondition(TRACKER_3_VARBITID, 1);
+
+		isUpstairsTower = new ZoneCondition(upstairsTower);
+
+		talkToFirstTracker = new Conditions(notCompleteFirstTracker);
+		talkToSecondTracker = new Conditions(LogicType.AND, completeFirstTracker, notCompleteSecondTracker);
+		talkToThirdTracker = new Conditions(LogicType.AND, completeFirstTracker, notCompleteThirdTracker);
+		shouldFireBallista = new Conditions(LogicType.AND, completeFirstTracker, completeSecondTracker, completeThirdTracker);
+	}
+
+	private void setupSteps()
+	{
+		talkToKingBolren = new NpcStep(this, NpcID.KING_BOLREN, new WorldPoint(2541, 3170, 0), "Speak to King Bolren in the centre of the Tree Gnome Maze.");
+		talkToKingBolren.addDialogStep("Can I help at all?");
+		talkToKingBolren.addDialogStep("I would be glad to help.");
+
+		talkToCommanderMontai = new NpcStep(this, NpcID.COMMANDER_MONTAI, new WorldPoint(2523, 3208, 0), "Speak with Commander Montai.");
+		talkToCommanderMontai.addDialogStep("Ok, I'll gather some wood.");
+
+		bringWoodToCommanderMontai = new NpcStep(this, NpcID.COMMANDER_MONTAI, new WorldPoint(2523, 3208, 0), "Speak with Commander Montai again to give him the wood.");
+
+		talkToCommanderMontaiAgain = new NpcStep(this, NpcID.COMMANDER_MONTAI, new WorldPoint(2523, 3208, 0), "Speak with Commander Montai.");
+		talkToCommanderMontaiAgain.addDialogStep("I'll try my best.");
+
+		firstTracker = new NpcStep(this, NpcID.TRACKER_GNOME_1, new WorldPoint(2501, 3261, 0), "Talk to the first tracker gnome to the northwest.");
+		secondTracker = new NpcStep(this, NpcID.TRACKER_GNOME_2, new WorldPoint(2524, 3257, 0), "Talk to the second tracker gnome inside the jail.");
+		thirdTracker = new NpcStep(this, NpcID.TRACKER_GNOME_3, new WorldPoint(2497, 3234, 0), "Talk to the third tracker gnome to the southwest.");
+
+		fireBallista = new ObjectStep(this, ObjectID.BALLISTA, new WorldPoint(2509, 3211, 0), "Fire the ballista.");
+
+		climbTheLadder = new ObjectStep(this, ObjectID.LADDER_16683, new WorldPoint(2503, 3252, 0), "Climb the ladder");
+
+		ItemRequirement firstOrb = new ItemRequirement("Orb of protection", ItemID.ORB_OF_PROTECTION, 1);
+		firstOrb.setTip("If you have lost the orb you can get another from the chest");
+		talkToKingBolrenFirstOrb = new NpcStep(this, NpcID.KING_BOLREN, new WorldPoint(2541, 3170, 0),
+			"Speak to King Bolren in the centre of the Tree Gnome Maze.",
+			firstOrb);
+		talkToKingBolrenFirstOrb.addDialogStep("I will find the warlord and bring back the orbs.");
+
+		returnOrbs = new NpcStep(this, NpcID.KING_BOLREN, new WorldPoint(2541, 3170, 0), "Speak to King Bolren in the centre of the Tree Gnome Maze.");
```

Comment:
This step should require the two orbs as an ItemRequirement.

Reply:
A step to prompt the player to pick up the two dropped orbs would be good. I believe also if the player doesn't pick them up they need to kill another khazard warlord, so maybe a tip for the orbs saying so would be good, or even just including it as part of this ConditionalStep. Using an ItemStep will work for telling the player to pick up the orbs, and you can use an ItemConditional to detect if the orbs are nearby.

Reply:
For the above, varbit 598 goes from 0 to 1 to represent you've handed in the orbs, but not finished the quest, so a separate step for that which overrides the above logic would be great.

Reply:
The ritual was autodone for me, are you saying that they can cancel it?

Reply:
Added everything but the ritual stuff.